pyresample/geometry.py

```diff
@@ -475,6 +475,57 @@ def __str__(self):
                                                     str(self.lons),
                                                     str(self.lats))
 
+    def geocentric_resolution(self, ellps='WGS84', radius=None, nadir_factor=2):
+        """Calculate maximum geocentric pixel resolution.
```

Comment:
This seems to be tailored for swath data, should be maybe move this to the `SwathDefinition` class instead ?

Reply:
Some people still use `GridDefinition` (which I would argue is a special case of the AreaDefinition) and so I put this here so it would work for Grid and Swath definitions.